aya-obj/src/links.rs

```diff
@@ -0,0 +1,100 @@
+//! Link type bindings.
+
+use crate::{
+    generated::{bpf_attach_type, bpf_link_type},
+    InvalidTypeBinding,
+};
+
+impl TryFrom<u32> for bpf_link_type {
+    type Error = InvalidTypeBinding<u32>;
+
+    fn try_from(link_type: u32) -> Result<Self, Self::Error> {
+        use bpf_link_type::*;
+        Ok(match link_type {
+            x if x == BPF_LINK_TYPE_UNSPEC as u32 => BPF_LINK_TYPE_UNSPEC,
```

Comment:
Kind of a bit nit here but I would instead transform link_type to `bpf_link_type::Type` that would remove all the as u32 boilerplate around :smile:

Reply:
Could you elaborate on `::Type`? I'm not familiar with the usage of that syntax. 🤔

Reply:
Those `BPF_LINK_TYPE_` values have a type of `bpf_link_type::Type` (that you can find defined [here for x86_64](https://github.com/aya-rs/aya/blob/635ed3baed5442c1364a360d7234b72c4ffe3fd8/ebpf/aya-ebpf-bindings/src/x86_64/bindings.rs#L698))

So the idea would be to cast link_type to `bpf_link_type::Type` instead of casting each  `BPF_LINK_TYPE_` values to u32.